### DISCUSSION TITLE
Docker运行报错

### DISCUSSION THREAD

**Discussion by DreamPWJ** [timonwong/prometheus-webhook-dingtalk#168] [Q&A]
执行命令
docker pull timonwong/prometheus-webhook-dingtalk
docker run -d --restart=always -p 8060:8060 --name prometheus-webhook-dingtalk timonwong/prometheus-webhook-dingtalk \
 --ding.profile="webhook1=https://oapi.dingtalk.com/robot/send?access_token=7e0a34d57be41808ab02b1955ed2f19d64d1fbd95e521331eff8cfe16e05b861"
报错信息
prometheus-webhook-dingtalk: error: path 'config.yml' does not exist, try --help

**Comment by timonwong** [comment]
你这个用法是 2.0.0 之前的版本。新版本已不做支持

https://github.com/timonwong/prometheus-webhook-dingtalk/blob/main/CHANGELOG.md

**Comment by DreamPWJ** [comment]
Configuration 文档不明确啊  这是要配置在那个文件内，docker内放在哪？

**Comment by timonwong** [comment]
不知道的话，用老版本吧

**Comment by DreamPWJ** [comment]
@timonwong   建议稍微完善一下文档， 我看issues都是说文档太简陋了。 毕竟开源项目除了代码，最重要的就是文档了，这是使用的入口， 毕竟小白比较多，如果都是大神的话就自己造一个了

**Comment by timonwong** [comment]
文档这个真的不好意思，现在已经不用钉钉了，所以也没有什么动力去完善文档，之前的博客文章也算过时很久了。
https://github.com/timonwong/prometheus-webhook-dingtalk/blob/main/docs/FAQ_zh.md
算是一个尝试，不过可以看到里面只有「点」而没有拉通整个「Getting Started」的这个「面」的相应教程。

不过可以告诉你的是，如果用 docker，需要利用 docker volume mount
比如
docker run -d --restart=always -p 8060:8060 -v
$PWD/config.yaml:/config/config.yaml --name prometheus-webhook-dingtalk
timonwong/prometheus-webhook-dingtalk --config.file=/config/config.yaml

docker run -v 参数指定挂载:
$PWD/config.yaml 为你的本地配置文件路径（当然你可以自己改）
/config/config.yaml 为挂载到 docker 容器里面的配置文件路径（这个也可以改），后面 --config.file
当然也需要同步修改路径

**Comment by timonwong** [comment]
启动后，会看到如下信息，包含 "Webhook urls for prometheus alertmanager" 文本的:

```
ts=2021-11-10T03:00:35.272Z caller=main.go:114 component=configuration msg="Webhook urls for prometheus alertmanager" urls="http://localhost:8060/dingtalk/webhook1/send http://localhost:8060/dingtalk/webhook2/send http://localhost:8060/dingtalk/webhook-test/send"
```

后面的urls 就是你需要在 alertmanager 里面配置的 webhook 地址，以空格隔开（当然 localhost:8060 需要自己调整、替换） 

上面 urls 里面的 webhook1, webhook2, webhook-test 就对应配置文件里面的 targets:

![image](https://user-images.githubusercontent.com/1398228/141042035-e54152a7-493e-4277-b14b-bea4989c9176.png)

**Comment by timonwong** [comment]
@DreamPWJ 我转换成了 Discussion，方便讨论点

**Comment by chenzq1604** [comment]
docker run -d --restart always -p 8060:8060 -v /opt/prometheus/plugs/config.yml:/etc/prometheus-webhook-dingtalk/config.yml timonwong/prometheus-webhook-dingtalk --web.listen-address="192.168.43.128:8060"

运行这个出错，不知道如何修改 localhost:8060

**Reply by chenzq1604** [reply]
可以了，把上面的修改去掉引号，再把config.yml引用就可以了
docker run -d --restart always -p 8060:8060 -v /opt/prometheus/plugs/config.yml:/etc/prometheus-webhook-dingtalk/config.yml \
timonwong/prometheus-webhook-dingtalk  --config.file=/etc/prometheus-webhook-dingtalk/config.yml --web.listen-address=0.0.0.0:8060
